### ISSUE TITLE
Possible struct_missing false positive

### ISSUE THREAD

**Issue by conradoplg** [obi1kenobi/cargo-semver-checks#467]
### Steps to reproduce the bug with the above code

Run `cargo semver-checks check-release --baseline-version 3.1.0` on commit d08ae22108f91ade84245b1a5f16a80c2e2ae255 (current `main`) of https://github.com/ZcashFoundation/ed25519-zebra

### Actual Behaviour

Fails with 

```
cargo semver-checks check-release
    Updating index
     Parsing ed25519-zebra v3.1.0 (current)
     Parsing ed25519-zebra v3.1.0 (baseline, cached)
    Checking ed25519-zebra v3.1.0 -> v3.1.0 (no change)
   Completed [   0.010s] 44 checks; 43 passed, 1 failed, 0 unnecessary

--- failure struct_missing: pub struct removed or renamed ---

Description:
A publicly-visible struct cannot be imported by its prior path. A `pub use` may have been removed, or the struct itself may have been renamed or removed entirely.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-check/tree/v0.21.0/src/lints/struct_missing.ron

Failed in:
  struct ed25519_zebra::Signature, previously in file /home/conrado/.cargo/registry/src/github.com-1ecc6299db9ec823/ed25519-zebra-3.1.0/src/signature.rs:8
       Final [   0.010s] semver requires new major version: 1 major and 0 minor checks failed
```

### Expected Behaviour

I don't think it should give this error. Before, the struct was a `pub use` for an internal struct:

https://github.com/ZcashFoundation/ed25519-zebra/blob/612e51af2e7eaa228f9a08ad38b0b0660e510d9e/src/lib.rs#L18

And now it's for a similar struct in an external crate, but it still exists:

https://github.com/ZcashFoundation/ed25519-zebra/blob/d08ae22108f91ade84245b1a5f16a80c2e2ae255/src/lib.rs#L19

Feels like I'm missing something obvious, sorry if that's the case

#### Software version

cargo-semver-checks 0.21.0

#### Operating system

Linux 5.19.0-43-generic

#### Command-line

```bash
/home/conrado/.cargo/bin/cargo-semver-checks semver-checks --bugreport 
```

#### cargo version

```
> cargo -V 
cargo 1.70.0 (ec8a8a0ca 2023-04-25)
```

#### Compile time information

- Profile: release
- Target triple: x86_64-unknown-linux-gnu
- Family: unix
- OS: linux
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-unknown-linux-gnu

**Comment by obi1kenobi** [comment]
You're right, this is an instance of the biggest class of false-positive we haven't been able to tackle yet. The overall problem is tracked by #355 so that's the best issue to watch — I'll close this issue in favor of that one.

TL;DR of why this happens: there's currently no reliable way to use rustdoc to resolve imports across crates. The semver analysis only looks at the crate being scanned, and doesn't see any dependencies' data — so moving and re-exporting an item from another crate looks like a deletion.

This is a problem for other use cases as well, not just cargo-semver-checks, so we've been working with the rustdoc team on a solution. We have an idea that should work but it isn't ready to ship yet. I can suggest a few people to sponsor on GitHub if you're able to do so and would like to help it along.
